src/vsr/replica.zig

```diff
@@ -1540,6 +1631,7 @@ pub fn ReplicaType(
 
             assert(!self.solo());
             assert(self.status == .view_change);
+            assert(self.sync_stage == .not_syncing);
```

Comment:
I started out that way, but it makes logging the current sync stage annoying. Avoiding the optional means we can always log `@tagName(self.sync_stage)`.

Reply:
What if we had a helper to ease logging the tag name?

As I was reading further, for example where we switch on syncing stage, it did seem like it would be worthwhile.

Reply:
Imo switching on the stage actually benefits from the stage being non-optional, too. Unfortunately zig doesn't allow `switch (foo) { null => ..., .a => ..., etc }`.

Reply:
I was thinking `switch (self.syncing.?) {}` given that the `not_syncing` branch is often unreachable. For example, in `on_sync_message_timeout`.

The assertions for the common case of not syncing will also read nicely, in the positive:

`assert(self.syncing == null)` vs
`assert(self.sync_stage != .not_syncing)`

(my prior suggestion for `not_syncing` was not great I think)